stun/stun.go

```diff
@@ -94,20 +95,22 @@ func HolePunch(portToStun, proto int) (publicIP net.IP, publicPort int, natType
 	}
 
 	for _, stunServer := range StunServers {
-		stunServer := stunServer
-		var err error
+		var err4 error
+		var err6 error
 		if proto == 4 {
-			publicIP, publicPort, natType, err = callHolePunch(stunServer, portToStun, "udp4")
-			if err != nil {
-				slog.Warn("callHolePunch udp4 error", err.Error())
+			publicIP, publicPort, natType, err4 = callHolePunch(stunServer, portToStun, "udp4")
+			if err4 != nil {
+				slog.Warn("callHolePunch udp4 error", err4.Error())
 			}
 		} else {
-			publicIP, publicPort, natType, err = callHolePunch(stunServer, portToStun, "udp6")
-			if err != nil {
-				slog.Warn("callHolePunch udp6 error", err.Error())
-				continue
+			publicIP, publicPort, natType, err6 = callHolePunch(stunServer, portToStun, "udp6")
+			if err6 != nil {
+				slog.Warn("callHolePunch udp6 error", err6.Error())
 			}
 		}
+		if err4 != nil || err6 != nil {
```

Comment:
why was this check changed to OR?
if a machine only has an ipv4 endpoint won't err6 be non-nil?

Reply:
ipv4 and ipv6 call will not be called at the same time.   For single stack vm, there is always one (err4 or err6 nil). 
On a dual stack vm,  if public ipv4 is not returned,   it will not try next stun server if with &&. 
|| should work in all kinds of situation.   Or it has to revert back to original code.